### PR TITLE
add: passwordless auth to app

### PR DESCRIPTION
**Pull request by lvancraen**
Passwordless auth has been added to the app. It can be deployed and still used with credentials. The next steps will be to do the tenant migration.

Shortcut: [ch6604](https://app.shortcut.com/capable/story/6604/add-support-for-passwordless-login-to-capable-care)

Video in slack: https://capable-health.slack.com/archives/C024KBUSPPU/p1660075135458419?thread_ts=1659983502.651489&cid=C024KBUSPPU

Screen Shot 2022-08-10 at 12 09 35 PM

### CODE REVIEW

**Comment by SWHandyMan** [comment]
@lvancraen Is there a reason the environment variables names can't be something like "standard" and "passwordless"? I feel this would be a lot more readable and clear than "custom auth"?

**Review by ebilg — APPROVED** [review]

**Comment by lvancraen** [comment]
> @lvancraen Is there a reason the environment variables names can't be something like "standard" and "passwordless"? I feel this would be a lot more readable and clear than "custom auth"?

We can name it whatever we want. I named it custom_auth because that's the language used in Cognito. But for us, I would agree passwordless would make more sense. I'll update it.

**Comment by lvancraen** [comment]
I noticed that in Kariko we initialize the amplify sdk using the url. Which means that we may not even need to configure it using a flag ... aka we wouldn't need to re-deploy it but perhaps simply have a different url. 1) uses the credentials flow and 2) uses the passwordless flow.

**Review by SWHandyMan — APPROVED** [review]

**Comment by lp** [comment]
> I noticed that in Kariko we initialize the amplify sdk using the url. Which means that we may not even need to configure it using a flag ... aka we wouldn't need to re-deploy it but perhaps simply have a different url. 1) uses the credentials flow and 2) uses the passwordless flow.

I'm not sure I follow @lvancraen, the _flow_ is not in the URL, only the `client_id` & `user_pool_id`

File: .env.sample [hunk 1]
```diff
@@ -3,7 +3,14 @@
...
 REACT_APP_COGNITO_USER_POOL_ID=<<xxxxxxxx>>
 # The ID below should be the End User Login App Client ID
-REACT_APP_COGNITO_USER_POOL_WEB_CLIENT_ID=<<xxxxxxxx>>
+REACT_APP_COGNITO_USER_POOL_WEB_CLIENT_ID_END_USER_LOGIN=<<xxxxxxxx>>
+# The ID below should be the Passwordless App Client ID
+REACT_APP_COGNITO_USER_POOL_WEB_CLIENT_ID_PASSWORDLESS=<<xxxxxxxx>>
+
+# Whether to show an auth flow (by default they're true)
+REACT_APP_ENABLE_CREDENTIALS=true
+REACT_APP_ENABLE_PASSWORDLESS=true
```
Comment (lp) [inline .env.sample:12]: what happens when both are `ON`?
are they mounded on different paths with some links to navigate between flows?

Comment (lvancraen) [reply .env.sample:12]: When they're both `ON`, the default is the credentials flow. However, you can change that by setting `REACT_APP_DEFAULT_AUTH_FLOW` to `passwordless`. You can also switch auth flow via the url by appending `passwordless` to the url. In the future, there will most likely be a button to allow users to navigate between flows. However, without a design, I didn't feel comfortable adding a button right now.

**Comment by lvancraen** [comment]
>I'm not sure I follow @lvancraen, the flow is not in the URL, only the client_id & user_pool_id
@lp I know. Which is why after a little more looking into it, I decided against using the url because it's only useful if we also code the clientIds in the code so as to figure out which auth flow it pertains too -- which I didn't like as a solution. I opted for something less flexible but more predictable.

**Review by stephanie-saretsky — APPROVED** [review]

**Review by lp — APPROVED** [review]

**Pull request merged**